ui/app/components/app/modals/customize-nonce/customize-nonce.component.js

```diff
@@ -0,0 +1,159 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import Modal from '../../modal';
+import TextField from '../../../ui/text-field';
+import Button from '../../../ui/button';
+import Typography from '../../../ui/typography';
+import {
+  COLORS,
+  TYPOGRAPHY,
+  FONT_WEIGHT,
+  ALIGN_ITEMS,
+  BLOCK_SIZES,
+} from '../../../../helpers/constants/design-system';
+import Box from '../../../ui/box';
+
+export default class CustomizeNonce extends PureComponent {
```

Comment:
One other pattern that I think we should try to ensure we follow moving forward: writing our components as functional components (not classes). I think you were copying existing patterns in this part of the code base which is fine, but so that we can more easily lift this out when we rework the confirmations in the future let's try and refactor this. 

Some notes

`contextTypes` has no corrolary in functional components, instead you can use the `useI18nContext` hook to get the `t` function.

State can be done using `useState` from React, and I recommend a single value instead of an object. Ex:
```js
const [customNonceValue, setCustomNonceValue] = useState('');
```

Your functions can just be declared in the function body. 

You can collapse the 'renderModalContent` jsx into the `render` jsx and make the return value of `render` be your return value of the method.

the .container.js pattern is also deprecated, instead you can use the redux `useSelector` hook and the selectors you're using in your `.container` to do the same thing but keep it inline with the component. 

Let me know if you need any more help with the refactor.